Generation failed with ClassNotFound com/squareup/kotlinpoet/TypeNames

Using this as configuration:

```xml
<plugin>
    <groupId>io.github.deweyjose</groupId>
    <artifactId>graphqlcodegen-maven-plugin</artifactId>
    <version>1.51.1</version>
    <executions>
        <execution>
            <goals>
                <goal>generate</goal>
            </goals>
        </execution>
    </executions>
    <configuration>
        <language>kotlin</language>
        <schemaPaths>
            <param>src/main/resources/schema/</param>
        </schemaPaths>
        <packageName>com.example.demo</packageName>
        <typeMapping>
            <BigDecimal>java.math.BigDecimal</BigDecimal>
        </typeMapping>
        <addGeneratedAnnotation>true</addGeneratedAnnotation>
        <addDeprecatedAnnotation>true</addDeprecatedAnnotation>
    </configuration>
</plugin>
```

The generation fails:
```
...
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  27.022 s
[INFO] Finished at: 2024-01-05T14:56:15+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate (default) on project app: Execution default of goal io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate failed: A required class was missing while executing io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate: com/squareup/kotlinpoet/TypeNames
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/demo/.m2/repository/io/github/deweyjose/graphqlcodegen-maven-plugin/1.51.1/graphqlcodegen-maven-plugin-1.51.1.jar
[ERROR] urls[1] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/codegen/graphql-dgs-codegen-core/6.1.1/graphql-dgs-codegen-core-6.1.1.jar
[ERROR] urls[2] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/1.9.21/kotlin-stdlib-1.9.21.jar
[ERROR] urls[3] = file:/Users/demo/.m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar
[ERROR] urls[4] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/codegen/graphql-dgs-codegen-shared-core/6.1.1/graphql-dgs-codegen-shared-core-6.1.1.jar
[ERROR] urls[5] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-reflect/1.9.21/kotlin-reflect-1.9.21.jar
[ERROR] urls[6] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/graphql-dgs/5.5.1/graphql-dgs-5.5.1.jar
[ERROR] urls[7] = file:/Users/demo/.m2/repository/com/graphql-java/graphql-java/19.2/graphql-java-19.2.jar
[ERROR] urls[8] = file:/Users/demo/.m2/repository/com/graphql-java/java-dataloader/3.2.0/java-dataloader-3.2.0.jar
[ERROR] urls[9] = file:/Users/demo/.m2/repository/org/reactivestreams/reactive-streams/1.0.3/reactive-streams-1.0.3.jar
[ERROR] urls[10] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.13.4/jackson-annotations-2.13.4.jar
[ERROR] urls[11] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.13.4/jackson-databind-2.13.4.jar
[ERROR] urls[12] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.13.4/jackson-core-2.13.4.jar
[ERROR] urls[13] = file:/Users/demo/.m2/repository/com/squareup/javapoet/1.13.0/javapoet-1.13.0.jar
[ERROR] urls[14] = file:/Users/demo/.m2/repository/com/squareup/kotlinpoet/1.15.3/kotlinpoet-1.15.3.jar
[ERROR] urls[15] = file:/Users/demo/.m2/repository/com/github/ajalt/clikt/clikt/4.2.1/clikt-4.2.1.jar
[ERROR] urls[16] = file:/Users/demo/.m2/repository/com/github/ajalt/mordant/mordant/2.2.0/mordant-2.2.0.jar
[ERROR] urls[17] = file:/Users/demo/.m2/repository/com/github/ajalt/colormath/colormath/3.3.1/colormath-3.3.1.jar
[ERROR] urls[18] = file:/Users/demo/.m2/repository/org/jetbrains/markdown/0.5.2/markdown-0.5.2.jar
[ERROR] urls[19] = file:/Users/demo/.m2/repository/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar
[ERROR] urls[20] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-common/1.9.10/kotlin-stdlib-common-1.9.10.jar
[ERROR] urls[21] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-json/1.6.2/kotlinx-serialization-json-1.6.2.jar
[ERROR] urls[22] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-json-jvm/1.6.2/kotlinx-serialization-json-jvm-1.6.2.jar
[ERROR] urls[23] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-core-jvm/1.6.2/kotlinx-serialization-core-jvm-1.6.2.jar
[ERROR] urls[24] = file:/Users/demo/.m2/repository/ch/qos/logback/logback-classic/1.4.14/logback-classic-1.4.14.jar
[ERROR] urls[25] = file:/Users/demo/.m2/repository/ch/qos/logback/logback-core/1.4.14/logback-core-1.4.14.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : com.squareup.kotlinpoet.TypeNames
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate (default) on project app: Execution default of goal io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate failed: A required class was missing while executing io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate: com/squareup/kotlinpoet/TypeNames
-----------------------------------------------------
realm =    plugin>io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:/Users/demo/.m2/repository/io/github/deweyjose/graphqlcodegen-maven-plugin/1.51.1/graphqlcodegen-maven-plugin-1.51.1.jar
urls[1] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/codegen/graphql-dgs-codegen-core/6.1.1/graphql-dgs-codegen-core-6.1.1.jar
urls[2] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/1.9.21/kotlin-stdlib-1.9.21.jar
urls[3] = file:/Users/demo/.m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar
urls[4] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/codegen/graphql-dgs-codegen-shared-core/6.1.1/graphql-dgs-codegen-shared-core-6.1.1.jar
urls[5] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-reflect/1.9.21/kotlin-reflect-1.9.21.jar
urls[6] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/graphql-dgs/5.5.1/graphql-dgs-5.5.1.jar
urls[7] = file:/Users/demo/.m2/repository/com/graphql-java/graphql-java/19.2/graphql-java-19.2.jar
urls[8] = file:/Users/demo/.m2/repository/com/graphql-java/java-dataloader/3.2.0/java-dataloader-3.2.0.jar
urls[9] = file:/Users/demo/.m2/repository/org/reactivestreams/reactive-streams/1.0.3/reactive-streams-1.0.3.jar
urls[10] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.13.4/jackson-annotations-2.13.4.jar
urls[11] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.13.4/jackson-databind-2.13.4.jar
urls[12] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.13.4/jackson-core-2.13.4.jar
urls[13] = file:/Users/demo/.m2/repository/com/squareup/javapoet/1.13.0/javapoet-1.13.0.jar
urls[14] = file:/Users/demo/.m2/repository/com/squareup/kotlinpoet/1.15.3/kotlinpoet-1.15.3.jar
urls[15] = file:/Users/demo/.m2/repository/com/github/ajalt/clikt/clikt/4.2.1/clikt-4.2.1.jar
urls[16] = file:/Users/demo/.m2/repository/com/github/ajalt/mordant/mordant/2.2.0/mordant-2.2.0.jar
urls[17] = file:/Users/demo/.m2/repository/com/github/ajalt/colormath/colormath/3.3.1/colormath-3.3.1.jar
urls[18] = file:/Users/demo/.m2/repository/org/jetbrains/markdown/0.5.2/markdown-0.5.2.jar
urls[19] = file:/Users/demo/.m2/repository/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar
urls[20] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-common/1.9.10/kotlin-stdlib-common-1.9.10.jar
urls[21] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-json/1.6.2/kotlinx-serialization-json-1.6.2.jar
urls[22] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-json-jvm/1.6.2/kotlinx-serialization-json-jvm-1.6.2.jar
urls[23] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-core-jvm/1.6.2/kotlinx-serialization-core-jvm-1.6.2.jar
urls[24] = file:/Users/demo/.m2/repository/ch/qos/logback/logback-classic/1.4.14/logback-classic-1.4.14.jar
urls[25] = file:/Users/demo/.m2/repository/ch/qos/logback/logback-core/1.4.14/logback-core-1.4.14.jar
Number of foreign imports: 1
import: Entry[import  from realm ClassRealm[maven.api, parent: null]]

-----------------------------------------------------

    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute (BuildCacheMojosExecutionStrategy.java:145)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate failed: A required class was missing while executing io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate: com/squareup/kotlinpoet/TypeNames
-----------------------------------------------------
realm =    plugin>io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:/Users/demo/.m2/repository/io/github/deweyjose/graphqlcodegen-maven-plugin/1.51.1/graphqlcodegen-maven-plugin-1.51.1.jar
urls[1] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/codegen/graphql-dgs-codegen-core/6.1.1/graphql-dgs-codegen-core-6.1.1.jar
urls[2] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/1.9.21/kotlin-stdlib-1.9.21.jar
urls[3] = file:/Users/demo/.m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar
urls[4] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/codegen/graphql-dgs-codegen-shared-core/6.1.1/graphql-dgs-codegen-shared-core-6.1.1.jar
urls[5] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-reflect/1.9.21/kotlin-reflect-1.9.21.jar
urls[6] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/graphql-dgs/5.5.1/graphql-dgs-5.5.1.jar
urls[7] = file:/Users/demo/.m2/repository/com/graphql-java/graphql-java/19.2/graphql-java-19.2.jar
urls[8] = file:/Users/demo/.m2/repository/com/graphql-java/java-dataloader/3.2.0/java-dataloader-3.2.0.jar
urls[9] = file:/Users/demo/.m2/repository/org/reactivestreams/reactive-streams/1.0.3/reactive-streams-1.0.3.jar
urls[10] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.13.4/jackson-annotations-2.13.4.jar
urls[11] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.13.4/jackson-databind-2.13.4.jar
urls[12] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.13.4/jackson-core-2.13.4.jar
urls[13] = file:/Users/demo/.m2/repository/com/squareup/javapoet/1.13.0/javapoet-1.13.0.jar
urls[14] = file:/Users/demo/.m2/repository/com/squareup/kotlinpoet/1.15.3/kotlinpoet-1.15.3.jar
urls[15] = file:/Users/demo/.m2/repository/com/github/ajalt/clikt/clikt/4.2.1/clikt-4.2.1.jar
urls[16] = file:/Users/demo/.m2/repository/com/github/ajalt/mordant/mordant/2.2.0/mordant-2.2.0.jar
urls[17] = file:/Users/demo/.m2/repository/com/github/ajalt/colormath/colormath/3.3.1/colormath-3.3.1.jar
urls[18] = file:/Users/demo/.m2/repository/org/jetbrains/markdown/0.5.2/markdown-0.5.2.jar
urls[19] = file:/Users/demo/.m2/repository/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar
urls[20] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-common/1.9.10/kotlin-stdlib-common-1.9.10.jar
urls[21] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-json/1.6.2/kotlinx-serialization-json-1.6.2.jar
urls[22] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-json-jvm/1.6.2/kotlinx-serialization-json-jvm-1.6.2.jar
urls[23] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-core-jvm/1.6.2/kotlinx-serialization-core-jvm-1.6.2.jar
urls[24] = file:/Users/demo/.m2/repository/ch/qos/logback/logback-classic/1.4.14/logback-classic-1.4.14.jar
urls[25] = file:/Users/demo/.m2/repository/ch/qos/logback/logback-core/1.4.14/logback-core-1.4.14.jar
Number of foreign imports: 1
import: Entry[import  from realm ClassRealm[maven.api, parent: null]]

-----------------------------------------------------

    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:152)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute (BuildCacheMojosExecutionStrategy.java:145)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
Caused by: org.apache.maven.plugin.PluginContainerException: A required class was missing while executing io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1:generate: com/squareup/kotlinpoet/TypeNames
-----------------------------------------------------
realm =    plugin>io.github.deweyjose:graphqlcodegen-maven-plugin:1.51.1
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:/Users/demo/.m2/repository/io/github/deweyjose/graphqlcodegen-maven-plugin/1.51.1/graphqlcodegen-maven-plugin-1.51.1.jar
urls[1] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/codegen/graphql-dgs-codegen-core/6.1.1/graphql-dgs-codegen-core-6.1.1.jar
urls[2] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/1.9.21/kotlin-stdlib-1.9.21.jar
urls[3] = file:/Users/demo/.m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar
urls[4] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/codegen/graphql-dgs-codegen-shared-core/6.1.1/graphql-dgs-codegen-shared-core-6.1.1.jar
urls[5] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-reflect/1.9.21/kotlin-reflect-1.9.21.jar
urls[6] = file:/Users/demo/.m2/repository/com/netflix/graphql/dgs/graphql-dgs/5.5.1/graphql-dgs-5.5.1.jar
urls[7] = file:/Users/demo/.m2/repository/com/graphql-java/graphql-java/19.2/graphql-java-19.2.jar
urls[8] = file:/Users/demo/.m2/repository/com/graphql-java/java-dataloader/3.2.0/java-dataloader-3.2.0.jar
urls[9] = file:/Users/demo/.m2/repository/org/reactivestreams/reactive-streams/1.0.3/reactive-streams-1.0.3.jar
urls[10] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.13.4/jackson-annotations-2.13.4.jar
urls[11] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.13.4/jackson-databind-2.13.4.jar
urls[12] = file:/Users/demo/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.13.4/jackson-core-2.13.4.jar
urls[13] = file:/Users/demo/.m2/repository/com/squareup/javapoet/1.13.0/javapoet-1.13.0.jar
urls[14] = file:/Users/demo/.m2/repository/com/squareup/kotlinpoet/1.15.3/kotlinpoet-1.15.3.jar
urls[15] = file:/Users/demo/.m2/repository/com/github/ajalt/clikt/clikt/4.2.1/clikt-4.2.1.jar
urls[16] = file:/Users/demo/.m2/repository/com/github/ajalt/mordant/mordant/2.2.0/mordant-2.2.0.jar
urls[17] = file:/Users/demo/.m2/repository/com/github/ajalt/colormath/colormath/3.3.1/colormath-3.3.1.jar
urls[18] = file:/Users/demo/.m2/repository/org/jetbrains/markdown/0.5.2/markdown-0.5.2.jar
urls[19] = file:/Users/demo/.m2/repository/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar
urls[20] = file:/Users/demo/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-common/1.9.10/kotlin-stdlib-common-1.9.10.jar
urls[21] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-json/1.6.2/kotlinx-serialization-json-1.6.2.jar
urls[22] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-json-jvm/1.6.2/kotlinx-serialization-json-jvm-1.6.2.jar
urls[23] = file:/Users/demo/.m2/repository/org/jetbrains/kotlinx/kotlinx-serialization-core-jvm/1.6.2/kotlinx-serialization-core-jvm-1.6.2.jar
urls[24] = file:/Users/demo/.m2/repository/ch/qos/logback/logback-classic/1.4.14/logback-classic-1.4.14.jar
urls[25] = file:/Users/demo/.m2/repository/ch/qos/logback/logback-core/1.4.14/logback-core-1.4.14.jar
Number of foreign imports: 1
import: Entry[import  from realm ClassRealm[maven.api, parent: null]]

-----------------------------------------------------

    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:150)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute (BuildCacheMojosExecutionStrategy.java:145)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
Caused by: java.lang.NoClassDefFoundError: com/squareup/kotlinpoet/TypeNames
    at com.netflix.graphql.dgs.codegen.generators.kotlin.KotlinTypeUtils.<init> (KotlinTypeUtils.kt:38)
    at com.netflix.graphql.dgs.codegen.generators.kotlin.AbstractKotlinDataTypeGenerator.<init> (KotlinDataTypeGenerator.kt:144)
    at com.netflix.graphql.dgs.codegen.generators.kotlin.KotlinDataTypeGenerator.<init> (KotlinDataTypeGenerator.kt:49)
    at com.netflix.graphql.dgs.codegen.CodeGen$generateKotlinDataTypes$3.invoke (CodeGen.kt:432)
    at com.netflix.graphql.dgs.codegen.CodeGen$generateKotlinDataTypes$3.invoke (CodeGen.kt:430)
    at kotlin.sequences.TransformingSequence$iterator$1.next (Sequences.kt:210)
    at com.netflix.graphql.dgs.codegen.CodeGen.generateKotlinDataTypes (CodeGen.kt:871)
    at com.netflix.graphql.dgs.codegen.CodeGen.generateKotlin (CodeGen.kt:331)
    at com.netflix.graphql.dgs.codegen.CodeGen.generate (CodeGen.kt:73)
    at io.github.deweyjose.graphqlcodegen.Codegen.execute (Codegen.java:260)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute (BuildCacheMojosExecutionStrategy.java:145)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
Caused by: java.lang.ClassNotFoundException: com.squareup.kotlinpoet.TypeNames
    at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass (SelfFirstStrategy.java:50)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass (ClassRealm.java:271)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:247)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:239)
    at com.netflix.graphql.dgs.codegen.generators.kotlin.KotlinTypeUtils.<init> (KotlinTypeUtils.kt:38)
    at com.netflix.graphql.dgs.codegen.generators.kotlin.AbstractKotlinDataTypeGenerator.<init> (KotlinDataTypeGenerator.kt:144)
    at com.netflix.graphql.dgs.codegen.generators.kotlin.KotlinDataTypeGenerator.<init> (KotlinDataTypeGenerator.kt:49)
    at com.netflix.graphql.dgs.codegen.CodeGen$generateKotlinDataTypes$3.invoke (CodeGen.kt:432)
    at com.netflix.graphql.dgs.codegen.CodeGen$generateKotlinDataTypes$3.invoke (CodeGen.kt:430)
    at kotlin.sequences.TransformingSequence$iterator$1.next (Sequences.kt:210)
    at com.netflix.graphql.dgs.codegen.CodeGen.generateKotlinDataTypes (CodeGen.kt:871)
    at com.netflix.graphql.dgs.codegen.CodeGen.generateKotlin (CodeGen.kt:331)
    at com.netflix.graphql.dgs.codegen.CodeGen.generate (CodeGen.kt:73)
    at io.github.deweyjose.graphqlcodegen.Codegen.execute (Codegen.java:260)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
---
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
```

I tried adding https://mvnrepository.com/artifact/com.squareup/kotlinpoet-metadata-specs as dependency (multiple versions), but that creates other errors:
```
---
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
Caused by: java.lang.NoSuchMethodError: 'java.nio.file.Path com.squareup.kotlinpoet.FileSpec.writeTo(java.nio.file.Path)'
    at com.netflix.graphql.dgs.codegen.CodeGen.generate (CodeGen.kt:90)
    at io.github.deweyjose.graphqlcodegen.Codegen.execute (Codegen.java:260)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
---
```

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
     <groupId>io.github.deweyjose</groupId>
     <artifactId>graphqlcodegen-maven-plugin</artifactId>
     <packaging>maven-plugin</packaging>
-    <version>1.51.1</version>
+    <version>1.51.2</version>
 
     <name>GraphQL Code Generator</name>
     <description>Maven port of the Netflix DGS GraphQL Codegen gradle build plugin</description>
@@ -51,9 +51,15 @@
         <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>
         <junit.version>5.10.1</junit.version>
         <lombok.version>1.18.28</lombok.version>
+        <kotlinpoet-jvm.version>1.15.3</kotlinpoet-jvm.version>
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>com.squareup</groupId>
+            <artifactId>kotlinpoet-jvm</artifactId>
+            <version>${kotlinpoet-jvm.version}</version>
+        </dependency>
         <dependency>
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-plugin-api</artifactId>

